lib/compress/zstd_compress_internal.h

```diff
@@ -93,6 +97,22 @@ typedef struct {
     U32  staticPrices;           /* prices follow a pre-defined cost structure, statistics are irrelevant */
 } optState_t;
 
+typedef struct {
+    BYTE const* nextSrc;    /* next block here to continue on current prefix */ /* TODO: Does this belong here? */
```

Comment:
probably not.
This information is never used by matchfinder.

It's my understanding that `ZSTD_matchState_t` is only for core match finder / parser capabilities,
while `ZSTD_blockState_t` preserves additional data that needs to be transferred between blocks.
`nextSrc` seems to belong there.

Reply:
Currently `ZSTD_blockState_t` transfers state from the last **compressed** block (or dictionary) to the next compressed block. Specifically, there are two copies `prevBlock` and `nextBlock`, and we only set `prevBlock = nextBlock` after we confirm that we are compressing the block. That isn't the case for `nextSrc`, since it needs to be updated for uncompressed blocks as well.

The logic for placing it here is that it is only used when `base`, `dictBase`, and the other variables here are updated. Given `ip` and `nextSrc`, we can update the rest of the variables. It could also go in the block state, which will then have to have a substructure for the part that needs "confirmation".

Reply:
Maybe there is a need to distinguish inter-blocks state, from inter-compressed-blocks state.
I guess the inter-compressed-blocks state would become a part of inter-block state, and could use the swapping strategy proposed here.